src/gadgets/nonnative/bignat.rs

```diff
@@ -438,6 +454,7 @@ impl<Scalar: PrimeField> BigNat<Scalar> {
     &self,
     mut cs: CS,
     other: &Self,
+    always_equal: &AllocatedBit,
```

Comment:
Making this a `Boolean` and performing the conversion from `AllocatedBit` allows the function to skip the additional constraint when `always_equal` is a `Boolean::Constant`.